test/functional/plugin/lsp_spec.lua

```diff
@@ -831,6 +831,29 @@ describe('LSP', function()
         'å ä ɧ 汉语 ↥ 🤦 🦄';
       }, buf_lines(1))
     end)
+    it('applies edits with last line after vim\'s last line', function()
+      local edits = {
+        make_edit(0, 0, 5, 0, {"All replaced"});
+      }
+      exec_lua('vim.lsp.util.apply_text_edits(...)', edits, 1)
+      eq({
+        'All replaced';
+      }, buf_lines(1))
+      edits = {
+        make_edit(0, 0, 6, 0, {"All replaced"});
+      }
+      exec_lua('vim.lsp.util.apply_text_edits(...)', edits, 1)
+      eq({
+        'All replaced';
+      }, buf_lines(1))
+      edits = {
+        make_edit(0, 0, 6, 2, {"All replaced"});
+      }
+      exec_lua('vim.lsp.util.apply_text_edits(...)', edits, 1)
+      eq({
+        'All replaced';
+      }, buf_lines(1))
+    end)
```

Comment:
It seems like each of these should be in their own little `it(...)` block. Otherwise they are changing the same buffer each time.

Each can have it's own little name why you're trying to do it `it('should apply edits to one extra row')`, `it('should apply edits to two extra rows')`, etc.

Reply:
Fixed, and wrapped in a describe block to better give context